### PR TITLE
ovirt_disk: automatically detect virtual size of qcow image

### PR DESCRIPTION
**Pull request by mnecas**
https://bugzilla.redhat.com/show_bug.cgi?id=1883949

### CODE REVIEW

**Comment by mnecas** [comment]
ci test please

**Review by mwperina — COMMENTED** [review]
Looks good to me, but we should have a review from storage team. Eyal , could you please take a look?

@shenitzky

**Comment by jekader** [comment]
ci test please

**Comment by mnecas** [comment]
ci test please

File: plugins/modules/ovirt_disk.py [hunk 1]
```diff
@@ -589,7 +594,16 @@ def build_entity(self):
             ) if logical_unit else None,
         )
         if hasattr(disk, 'initial_size') and self._module.params['upload_image_path']:
-            disk.initial_size = size
+            out = subprocess.check_output([
+                'qemu-img',
+                'measure',
+                '-f', 'qcow2' if self._module.params.get('format') == 'cow' else 'raw',
```
Comment (nirs) [inline plugins/modules/ovirt_disk.py:600]: qemu-img measure can detect image format, just like qemu-img info above.

File: plugins/modules/ovirt_disk.py [hunk 2]
```diff
@@ -362,6 +362,8 @@
 import time
 import traceback
 import ssl
+import json
+import subprocess
```
Comment (nirs) [inline plugins/modules/ovirt_disk.py:366]: Need to sort the imports

File: plugins/modules/ovirt_disk.py [hunk 3]
```diff
@@ -589,7 +594,16 @@ def build_entity(self):
...
+            out = subprocess.check_output([
+                'qemu-img',
+                'measure',
+                '-f', 'qcow2' if self._module.params.get('format') == 'cow' else 'raw',
+                '-O', 'qcow2' if self._module.params.get('format') == 'cow' else 'raw',
+                '--output', 'json',
+                self._module.params['upload_image_path']
+            ])
+            measure = json.loads(out)
+            disk.initial_size = measure["required"]
```
Comment (barpavel) [inline plugins/modules/ovirt_disk.py:605]: measure["required"] is not a boolean value?
disk.initial_size shouldn't be a number?

Comment (nirs) [reply plugins/modules/ovirt_disk.py:605]: Did you try to run qemu-img measure? It will answer your question.

Comment (barpavel) [reply plugins/modules/ovirt_disk.py:605]: OK. No issues :)
The JSON output is an object of QAPI type BlockMeasureInfo
Fields
fully_allocated: isize
required: isize

**Comment by barpavel** [comment]
LGTM :+1:

**Review by mwperina — APPROVED** [review]
+1

**Pull request merged**